lasagne/layers/recurrent.py

```diff
@@ -153,61 +177,74 @@ def step_masked(layer_input, mask, hid_previous):
             sequences = input
             step_fun = step
 
-        output = theano.scan(step_fun, sequences=sequences,
-                             go_backwards=self.backwards,
-                             outputs_info=[self.hid_init],
-                             truncate_gradient=self.gradient_steps)[0]
+        # repeat hid_init to batch size
```

Comment:
I think computing a dot product here is more expensive than using the broadcastable attribute of hid_init.  Can we do that instead?